segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DebugDataFrameReader.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.logs;
+
+import io.pravega.common.Exceptions;
+import io.pravega.segmentstore.contracts.SequencedElement;
+import io.pravega.segmentstore.server.DataCorruptionException;
+import io.pravega.segmentstore.storage.DurableDataLog;
+import io.pravega.segmentstore.storage.DurableDataLogException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+
+@Slf4j
+public class DebugDataFrameReader<T extends SequencedElement> extends DataFrameReader<T> {
+
+    DebugDataFrameReader(DurableDataLog log, Serializer<T> serializer, int containerId) throws DurableDataLogException {
+       super(log, serializer, containerId);
+    }
+
+    /**
+     * This overridden getNext does not throw a DataCorruptionException on
+     * finding duplicate entries.
+     */
+    @Override
+    public DataFrameRecord<T> getNext() throws DataCorruptionException, DurableDataLogException {
+        Exceptions.checkNotClosed(this.isClosed(), this.isClosed());
+
+        try {
+            while (!this.getDataFrameInputStream().isClosed()) {
+                try {
+                    if (!this.getDataFrameInputStream().beginRecord()) {
+                        // We've reached the end of the DataFrameInputStream.
+                        return null;
+                    }
+                    // Attempt to deserialize the next record. If the serialization was bad, this will throw an exception which we'll pass along.
+                    T logItem = this.getSerializer().deserialize(this.getDataFrameInputStream());
+                    DataFrameRecord.RecordInfo recordInfo = this.getDataFrameInputStream().endRecord();
+                    long seqNo = logItem.getSequenceNumber();
+                    if (seqNo <= this.getLastReadSequenceNumber()) {
+                        log.error("Invalid Operation Sequence Number. Expected: larger than {}, found: {}.",
```

Comment:
I thin this should be using an `if` and `else` block. If we find a duplicate, we log the error and that's it, we just discard the item read. In the case it is not duplicate, then we return the dataframe. See this https://github.com/RaulGracia/pravega/commit/b763701221e00620042c5fd63d1f6f7fe9bab63a